File: AndroidBasics/notes/src/main/java/pan/alexander/notes/di/ApplicationComponent.java
package pan.alexander.notes.di;

import javax.inject.Named;
import javax.inject.Singleton;

import dagger.Component;
import dagger.Lazy;
import pan.alexander.notes.data.database.NotesDao;
import pan.alexander.notes.data.database.TrashDao;
import pan.alexander.notes.domain.MainInteractor;

@Singleton
@Component(modules = {RepositoryModule.class, RoomModule.class})
public interface ApplicationComponent {

    Lazy<NotesDao> getNotesDao();

    Lazy<TrashDao> getTrashDao();

    Lazy<MainInteractor> getMainInteractor();
}

File: Filmrevealer/training/src/main/java/pan/alexander/training/di/ApplicationComponent.kt
package pan.alexander.training.di

import dagger.Component
import pan.alexander.training.domain.MainInteractor

@Component(
    modules = [ContentResolverModule::class, RepositoryModule::class,
        CoroutinesModule::class, LocationModule::class, ContextModule::class]
)
interface ApplicationComponent {
    fun getMainInteractor(): dagger.Lazy<MainInteractor>
}
